x/evm/precompile/wasm.go

```diff
@@ -62,6 +65,12 @@ func (p precompileWasm) Run(
 	if err != nil {
 		return nil, err
 	}
+
+	// Gas consumed by a local gas meter
+	gasUsed := startResult.CacheCtx.GasMeter().GasConsumed()
+	if !contract.UseGas(gasUsed) {
+		return nil, vm.ErrOutOfGas
```

Comment:
Pretty sure this will never run either, because if `gasUsed` > `contract.Gas`, then the panic error case would've been handled in the deferred `ReturnToParentGasMeter` already.